### DISCUSSION TITLE
Displaying Last Updated Dates for Blog Posts?

### DISCUSSION THREAD

**Discussion by ToryDeng** [alshedivat/al-folio#2309] [Ideas]
Hi,

Thank you for your awesome template! 

Currently, each blog post only includes a single `date` attribute that represents the creation date and is added manually. I recently updated one of my older blog posts and it could be beneficial to display both the original creation date and the most recent update date. An example would look something like:

```text
# Post Title
Created on December 08, 2023   Last updated on April 01, 2024
```

I wonder if this feature is currently available or can be added to the template. Moreover, can the `last updated date` be automatically updated based on the modification time of the post's `.md` file? Sorry, I'm not very familiar with HTML and I don't know whether it can be implemented...

**Comment by george-gca** [comment]
I believe doing this would be too troublesome and would have some edge cases. For example, what if I want to manually set my last updated date? Should we use the last modified date from the file or the date from the commit (that might be different)?

I believe it is way easier to simply have a `last_updated` variable in the front matter of the post, and use that inside the post. This way you can manually set this date. This might be a good addition to the template. Would you mind sending a PR for this? You can base the implementation on the way that the date is currently handled.

**Reply by ToryDeng** [reply]
Sorry for the late reply; I've been busy these past weeks... I have submitted a pull request (#2341) to introduce an optional `last_updated` field. This is my first pull request, and I'm still familiarizing myself with web development practices, so I'm aware there could be some areas for improvement. I am very keen to address any issues or feedback you may have regarding my pull request. Thank you for your understanding and patience.
